Asterix/optics/propagation_functions.py

```diff
@@ -600,6 +604,8 @@ def prop_fpm_regional_sampling(pup, fpm, nbres=np.array([0.1, 5, 50, 100]), samp
     -------
     array : E-field before the Lyot stop.
     """
+    from Asterix.optics import shift_phase_ramp
```

Comment:
Yes ok. Since `import Asterix.optics.phase_amplitude_functions as phase_ampl` doesn't solve the circular import issue, I left things as they are and opened a new issue to move the function as you describe it.